quark/subproject.py

```diff
@@ -75,6 +75,8 @@ def _parse_fragment(url):
         res = {}
         for equality in url.fragment.split("&"):
             index = equality.find('=')
+            if index < 0:
```

Comment:
Just for me to understand: in what cases did this problem arise? Because I'm not sure that the correct thing here is to skip the sub-fragment, if the comment specifies that this arose from an empty dictionary value, probably the correct thing to do is to add an entry with the given key but an empty value.

Reply:
When there's no fragment, the resulting dict would be `{'': ''}`.

Reply:
After internal discussion I added an alternative implementation in d3cb7c1

Reply:
alternative implementation in [2884aca](https://github.com/comelz/quark/pull/51/commits/2884aca9828b5d23c82aee3f615b0fa12181321b), should almost match https://url.spec.whatwg.org/#urlencoded-parsing (except for the special treatment of `+`); empty chunks are skipped (so both the empty fragment and any sequence of multiple `&` produce nothing), chunks without an equal sign are interpreted as a key with no value.